wz_strategic_plugin/src/wz_strategic_plugin.cpp

```diff
@@ -0,0 +1,282 @@
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::STOP_THEN_PROCEED:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::STOP_AND_REMAIN:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PRE_MOVEMENT:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PERMISSIVE_MOVEMENT_ALLOWED:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PROTECTED_MOVEMENT_ALLOWED:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PERMISSIVE_CLEARANCE:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PROTECTED_CLEARANCE:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::CAUTION_CONFLICTING_TRAFFIC:
+                traffic_light_inter = "R";
+                break;
+        }
+
+        return traffic_light_inter;
+
+    }
+
+    bool WzStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp)
+    {
+        ROS_DEBUG("Finding car infomrmation");
+
+        // find car_down_track
+        double current_car_down_track = wm_->routeTrackPos(current_loc_).downtrack;
+        ROS_DEBUG("current_car_down_track %d", current_car_down_track);
+
+        auto current_lanelets = lanelet::geometry::findNearest(wm_->getMap()->laneletLayer, current_loc_, 1);
+        auto current_lanelet = current_lanelets[0];
+
+        ROS_DEBUG("\n\nFinding traffic_light information");
+        auto traffic_list = current_lanelet.second.regulatoryElementsAs<lanelet::CarmaTrafficLight>();
+
+        if(!traffic_list.empty()) {
+
+            auto nearest_traffic_light = traffic_list.front();
+
+            auto point = lanelet::traits::to2D(nearest_traffic_light->stopLine());
+            auto pos = carma_wm::geometry::trackPos(current_lanelet.second, point);
+            double traffic_light_down_track = pos.downtrack;
+
+            // double traffic_light_down_track = 0;
+            ROS_DEBUG("traffic_light_down_track %d", traffic_light_down_track);
+
+            double distance_remaining_to_traffic_light  = traffic_light_down_track - current_car_down_track;
+            ROS_DEBUG("distance_remaining_to_traffic_light %d", distance_remaining_to_traffic_light);
+
+            ROS_DEBUG("min_distance_to_traffic_light %d", min_distance_to_traffic_light);
+
+            ros::Time current_time = ros::Time::now();
+            ros::Time end_time = ros::Time::now();
+
+            if(distance_remaining_to_traffic_light >= min_distance_to_traffic_light ) {        
```

Comment:
Can we add an else condition to this as well? Maybe just a log statement that shows we're not entering this loop. If that means the strategic plugin cannot handle this condition, we should return a false and indicate failure